includes/admin.form.inc

```diff
@@ -71,6 +71,49 @@ function islandora_oai_admin_form($form, $form_state) {
     '#default_value' => variable_get('islandora_oai_expire_time', '86400'),
     '#description' => t('The amount of time a resumption token will remain valid, in seconds. Defaults to one day (86400s).'),
   );
+  $form['islandora_oai_configuration']['islandora_oai_query_backend'] = array(
+    '#type' => 'radios',
+    '#title' => t('Query Backend'),
+    '#default_value' => variable_get('islandora_oai_query_backend', 'sparql'),
+    '#description' => t('For larger repositories, OAI may perform poorly when attempting to perform the SPARQL queries it requires. In these cases, using the Solr backend may provide better results.'),
+    '#options' => array(
+      'sparql' => t('SPARQL'),
+      'solr' => t('Solr'),
+    ),
+  );
+  $form['islandora_oai_configuration']['islandora_oai_solr_state_field'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Solr Object State Field'),
+    '#default_value' => variable_get('islandora_oai_solr_state_field', 'fgs_state_s'),
+    '#description' => t("The field in Solr that holds a Fedora object's state. Must be a single-valued string type field."),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_oai_configuration[islandora_oai_query_backend]"]' => array('value' => 'solr'),
```

Comment:
Kind of nitpicky, but I would be inclined to pull this reused chunk out to a variable, to avoid repeating it for every dependent form element... Could be the whole array under `#states`, really?